week-8/Homework/mandatory/3-dog-photo-gallery/script.js

```diff
@@ -0,0 +1,19 @@
+let list = document.createElement("li");
+let image = document.createElement("img");
+
+let ulList = document.querySelector("#dog");
+ulList.appendChild(list);
+list.appendChild(image);
```

Comment:
This looks great :) An extra question to think about is _when_ to create elements and put them in the page. Imagine the `fetch` here failed, or took a really long time - it may be kind of weird to have an image element in the page with no source. It may be useful to not call `appendChild` until we have a `src` for the image (or maybe even to have a placeholder text saying "Loading..." which we replace with the image). This applies to your answer to the "humour" question too :)

Reply:
Actually, I just re-read the problem statement - each time you click it's meant to add a new dog to the list, rather than replacing each time (note the key word "append" in the question) - can you give that a go? :)